File: Naveen Babu Govindraj Naidu/Week1/mta-lab/mta.js
console.log("Mta-Lab");
// MTA-Lab:

  const N =["TS","34th","28th","23rd","US","8th"];
  const L =["8","6th","US","3rd","1st"];
  const S =["GC","33rd","28th","23","US","AP"];

  let noOfStops = 0;
  let stopsTravelled = [];
const journeyOfUser = function(startLine,getIn,endLine,getOff){

    let getInIndex = startLine.indexOf(getIn);
    let startInterIndex = startLine.indexOf("US");
    let endInterIndex = endLine.indexOf("US");
    let getOffIndex = endLine.indexOf(getOff);
    let totalStops1 = startInterIndex - getInIndex;
    let totalStops2 = Math.abs(endInterIndex-getOffIndex);
    noOfStops = totalStops1 + totalStops2;
    getInIndex = getInIndex + 1;
    for (i = 0; i < totalStops1; i++){
      stopsTravelled[i] = startLine[getInIndex];
      getInIndex++;
    }

    for(i = totalStops1 - 1;i < noOfStops - 1; i++){
      if(getOffIndex < endInterIndex){
        stopsTravelled[i] = endLine[endInterIndex];
        endInterIndex--;
      }else{
        stopsTravelled[i] = endLine[endInterIndex];
        endInterIndex++;
      }
    }
    console.log(`The total number of stops:${noOfStops -1}`);
    console.log(`You must travel through the following stops:${stopsTravelled}`);
    
  };
     function planTrip(startLine,getIn,endLine,getOff){
      if(startLine == "N" || "L" || "S" && endLine == "N" || "L" || "S"){
        journeyOfUser(startLine,getIn,endLine,getOff);
      }else{
        console.log("The line name is Incorrect!!!");
      }
    };
planTrip(S, 'GC', N, 'TS');

File: Naveen Babu Govindraj Naidu/Ruby/mta.rb
$lines = {
  'n' => ['TS','34th','28th','23rd','US','8th'],
  'l' => ['8','6th','US','3rd','1st'],
  'six' => ['GC','33rd','28','23rd','US','AP']
}

def plan_trip (start_line, get_in, end_line, get_off)

    getin_index = $lines[start_line].index(get_in)
    start_inter_index = $lines[start_line].index('US')
    end_inter_index = $lines[end_line].index('US')
    getoff_Index = $lines[end_line].index(get_off)
    stops1 = start_inter_index.to_i - (getin_index.to_i + 1)
    # puts stops1
    stops2 = (end_inter_index.to_i - getoff_Index.to_i).abs
    # puts stops2
    total_stops = stops1 + stops2
    puts total_stops
    stops = ''
    for i in get_off
      stops += i
      puts stops
    end






end
plan_trip('n','TS','n','8th')
